### PR TITLE
Document some undocumented configuration entries

### PR DESCRIPTION
**Pull request by PHBG-demant**
This change:

  * Documents the dog-config-path-resolve-symlink configuration entry. It appears to have been missed in prior documentation efforts.

  * Documents the dog-config-path and version configuration entries; these cannot be set but can be used in value interpolation. In the documentation, these types of entries are now described as constants.

  * Expands the documentation on value interpolation, noting that it is done after the effective configuration has been resolved. This means the include-dog-config entry cannot utilize value interpolation.

  * Sorts configuration entries alphabetically in the documentation.

  * Groups the dog.py string constants and sorts them alphabetically. This makes it easier to see what a constant is used for and thus determining whether more configuration entries lack documentation.

The ARGS configuration entry could also be documented as a constant with the consequence of making it part of the public API. It is unknown whether this is useful however, hence it has been ignored by this change.

### CODE REVIEW

**Comment by PHBG-demant** [comment]
BTW, after docker-compose functionality was removed it might make sense to bump the dog version number. It was a breaking change.

File: dog.py [hunk 1]
```diff
@@ -18,19 +18,15 @@
 MAX_DOG_CONFIG_VERSION = 2
 
 # Constants for consistent naming of dog variables, etc.
+# Configuration keys (documented, external)
```
Comment (rasmus-toftdahl-olesen) [inline dog.py:21]:
```suggestion
# Configuration keys (documented, externally in Configuration.md)
```

Comment (rasmus-toftdahl-olesen) [reply dog.py:21]: I think we should refer to the .md file with the documentation.

**Review by rasmus-toftdahl-olesen — CHANGES_REQUESTED** [review]
Hi Philip sorry for the slow review.

Just one suggestion, and then it is good to merge.

Really nice change by the way - and wonderful that someone has the skills to give the documentation some love - many projects are suffering on that account.

File: dog.py (hunk 1 above)
Comment (PHBG-demant) [reply dog.py]: Done

Comment (rasmus-toftdahl-olesen) [reply dog.py]: Cool, then I am very happy to merge it.

**Review by rasmus-toftdahl-olesen — APPROVED** [review]

**Pull request merged**